microceph/ceph/osd.go

```diff
@@ -1117,3 +1117,35 @@ func SetReplicationFactor(pools []string, size int64) error {
 
 	return nil
 }
+
+// GetOSDPools returns a list of OSD Pools and their configurations.
+func GetOSDPools() ([]types.Pool, error) {
+	out, err := processExec.RunCommand("ceph", "osd", "pool", "ls")
+	if err != nil {
+		return nil, fmt.Errorf("failed to list pools: %w", err)
+	}
+
+	poolNames := strings.Split(out, "\n")
```

Comment:
Although I agree that parsing the output is simple, still I would prefer this output be read with `--format json` and be unmarshal'd into a `[]string{}`.

what are your thoughts ?

Reply:
Makes sense, I've updated the call in `SetReplicationFactor` to also use this pattern.

Reply:
great :D